pkgs/development/python-modules/hydrus/default.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenv, fetchFromGitHub, lzma, qt5, wrapQtAppsHook
+, buildPythonPackage, pythonPackages, isPy3k, python
+, miniupnpc_2, swftools
+}:
+
+buildPythonPackage {
+  pname = "hydrus";
+  version = "413";
+  format = "other";
+  disabled = !isPy3k;
+
+  src = fetchFromGitHub {
+    owner = "hydrusnetwork";
+    repo = "hydrus";
+    rev = "9fbed11bef499e01a6799b298bea7d0967d30430";
+    sha256 = "1dl7qpzmlxl376lzm0chmwvf4nl55wz6fwcsw0ikb33rm8r33gq4";
+  };
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+  ];
+
+  propagatedBuildInputs = with pythonPackages; [
+    beautifulsoup4 html5lib lxml numpy opencv4 pillow psutil
+    pyopenssl pyyaml requests send2trash service-identity twisted
+    lz4 lzma pysocks matplotlib qtpy pyside2 
+  ];
+
+  # tests require a graphical session
+  doCheck = false;
```

Comment:
Not all of them require a graphical session right? It looks like there are at least some which just run against the server, maybe you can try running those?

Reply:
While a dry `nosetest` run shows `68/90` failing, I can only manage to get 11 passing when excluding the failing tests/modules/whatever.